### PR TITLE
Permutive ophan integration

### PR DESCRIPTION
**Pull request by adamnfish**
## What does this change?

Integrates the Ophan browser ID with Permutive. The permutive object includes an `identify` call that lets clients connect Permutive's internal ID up with their own IDs. This lets us identify users based on their Ophan browser ID.

## Does this change need to be reproduced in dotcom-rendering ?

- [ ] No
- [x] Yes (please indicate your plans for DCR Implementation)

Permutive is not yet set up on DCR, this change will be required when that is done.

## What is the value of this and can you measure success?

Permutive will be connected with our Browser ID, which is useful for reporting among other things.

### Does this affect other platforms?

No

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

This won't be in effect for ad-free (nor for users without advertising consent).

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/master/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

N/A, not user-facing.

### Tested

- [x] Locally
- [ ] On CODE (optional)

### CODE REVIEW

**Review by ripecosta — APPROVED** [review]
👍

File: static/src/javascripts/projects/commercial/modules/dfp/prepare-permutive.js [hunk 1]
```diff
@@ -100,8 +117,12 @@ const runPermutive = (
             throw new Error('Global Permutive setup error');
         }
 
-        const payload = generatePayload(pageConfig);
+        const permutiveIdentities = generatePermutiveIdentities(pageConfig);
+        if (permutiveIdentities.length > 0) {
+            permutiveGlobal.identify(permutiveIdentities);
```
Comment (GHaberis) [inline static/src/javascripts/projects/commercial/modules/dfp/prepare-permutive.js]: Should we check whether `permutiveGlobal.identify` is defined as we do for `permutiveGlobal.addon` elsewhere in this function?

Comment (adamnfish) [reply static/src/javascripts/projects/commercial/modules/dfp/prepare-permutive.js]: The previous block is checking that permutive is set up and providing a more helpful error if it is not, if it's there we expect it to be correct. I'm not sure we'd typically check the existence of all our third party module's methods before calling them, although doing so isn't without value in JavaScript.

Be great to chat to you in person to get your advice on the best way to implement a check that provides useful developer feedback.

File: static/src/javascripts/projects/commercial/modules/dfp/prepare-permutive.js [hunk 2]
```diff
@@ -90,6 +95,18 @@ const generatePayload = (
     return cleanPayload;
 };
 
+const generatePermutiveIdentities = (
+    pageConfig: Config = {}
+): Array<PermutiveIdentity> => {
+    if (
+        typeof pageConfig.ophan === 'object' &&
+        typeof pageConfig.ophan.browserId === 'string'
```
Comment (faresite) [inline static/src/javascripts/projects/commercial/modules/dfp/prepare-permutive.js]: Could this ever be an empty string?

Comment (adamnfish) [reply static/src/javascripts/projects/commercial/modules/dfp/prepare-permutive.js]: Good shout, I agree if this happens it should be excluded.

File: static/src/javascripts/projects/commercial/modules/dfp/prepare-permutive.js (hunk 1 above)
Comment (adamnfish) [reply static/src/javascripts/projects/commercial/modules/dfp/prepare-permutive.js]: We agreed this isn't required, but we should look at improving the existing check in a separate PR.

**Pull request merged**